universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
```

Comment:
It's not a "contract", but I find the name misleading as proposed.

The use of the vis parameter is particularly confusing in terms of its function and how it should be represented in the name, because it doesn't do what I'd expect, which is a test based on the visibility of the object, but rather is an independent parameter that controls whether to do anything at all. That would be clearer if tested at the call location, so that it doesn't appear from inspect of the calling code that something will happen which isn't the case.

I don't have a sense of what HandleShipDesignKnowledgeGrant will do from the name, though that's partly just that it isn't conforming to the vague conventions I'm expecting / imagining.